src/Tmds.ExecFunction/ExecFunction.cs

```diff
@@ -198,62 +202,76 @@ private static MethodInfo GetMethodInfo(Delegate d)
             return d.GetMethodInfo();
         }
 
-        static ExecFunction()
-        {
-            HostFilename = Process.GetCurrentProcess().MainModule.FileName;
-            string[] appArguments = null;
 
-            // application is running as 'testhost'
-            // try to find parent 'dotnet' host process.
-            if (HostFilename.EndsWith("/testhost") || HostFilename.EndsWith("\\testhost.exe"))
+        private static Lazy<ExecFunctionHostArgs> _execFunctiohHostArgsDefault = 
+            new Lazy<ExecFunctionHostArgs>(() => new ExecFunctionHostArgs(typeof(ExecFunction).Assembly.Location));
+
+        public class ExecFunctionHostArgs
```

Comment:
Can you remove `class ExecFunctionHostArgs`, and add `HostFilename`/`HostArguments` to `class ExecFunctionOptions` and use that in `ConfigureProcessStartInfoForMethodInvocation`.

nit: for consistency with `ProcessStartInfo`, use `HostFileName` (capital `N`).